Migration.Tool.Extensions/README.md

```diff
@@ -1,123 +1,78 @@
+- `ShallMigrate` - A boolean method that specifies whether the migration shall be applied for the current widget. Use properties of the `WidgetMigrationContext` and `WidgetIdentifier` objects passed as an argument to the method to evaluate the condition:
+  - `WidgetMigrationContext.SiteId` - An integer property that specifies the ID of the site on which the widget was used in the source instance.
+  - `WidgetIdentifier.TypeIdentifier` - A string property that specifies the identifier of the widget.
+- `MigrateWidget`- Migrate the widget data using the following properties:
+  - `identifier` - A `WidgetIdentifier` object, see `ShallMigrate` to see properties.
+  - `value` - A `JToken` object containing the deserialized value of the property.
+  - `context` - A `WidgetMigrationContext` object, see `ShallMigrate` to see properties.
+
+You can see a sample: [SampleWidgetMigration.cs](./CommunityMigrations/SampleWidgetMigration.cs)
+
+After implementing the migration, you need to [register the migration](#Register migrations) in the system.
+
+## Customize widget property migrations
+
+In the `Migration.Tool.Extensions/CommunityMigrations` folder, create a new file with a class that implements the `IWidgetPropertyMigration` interface. Implement the following properties and methods required by the interface:
+
+- `Rank` - An integer property that determines the order in which migrations are applied. Use a value lower than *100000*, as that is the value for system migrations.
+- `ShallMigrate` - A boolean method that specifies whether the migration shall be applied for the current widget. Use properties of the `WidgetPropertyMigrationContext` and `propertyName` objects passed as an argument to the method to evaluate the condition:
+  - `WidgetPropertyMigrationContext.SiteId` - An integer property that specifies the ID of the site on which the widget was used in the source instance.
+  - `WidgetPropertyMigrationContext.EditingFormControlModel` - An object representing the [form control](https://docs.kentico.com/x/lAyRBg) of the property.
+  - `propertyName` - A string property that specifies the identifier of the property.
+- `MigrateWidgetProperty`- Migrate the widget property data using the following properties:
+  - `key` - Name of the property.
+  - `value` - A `JToken` object containing the deserialized value of the property.
+  - `context` - A `WidgetPropertyMigrationContext`. See `ShallMigrate` method to see the properties.
+
+You can see samples:
+
 - [Path selector migration](./DefaultMigrations/WidgetPathSelectorMigration.cs)
 - [Page selector migration](./DefaultMigrations/WidgetPageSelectorMigration.cs)
-- [File selector migration](./DefaultMigrations/WidgetFileMigration.cs)
\ No newline at end of file
+- [File selector migration](./DefaultMigrations/WidgetFileMigration.cs)
+
+After implementing the migration, you need to [register the migration](#Register migrations) in the system.
```

Comment:
Do these link really work? I am not sure whether this is the correct format of the section link but in preview it seems to be broken.